packages/synapse-react-client/src/components/favorites/FavoritesPage.tsx

```diff
@@ -27,15 +35,84 @@ export type Sort = {
   direction: SortDirection
 }
 
+const columnHelper = createColumnHelper<EntityHeader>()
+
+const columns = [
+  columnHelper.display({
+    id: 'removeFavorite',
+    cell: ctx => <FavoriteButton entityId={ctx.row.original.id} />,
+    maxSize: 5,
+    enableResizing: false,
+  }),
+  columnHelper.accessor('name', {
+    cell: ctx => (
+      <a
+        rel="noopener noreferrer"
+        href={`${PRODUCTION_ENDPOINT_CONFIG.PORTAL}Synapse:${ctx.row.original.id}`}
+      >
+        {ctx.row.original.name}
+      </a>
+    ),
+    enableGlobalFilter: true,
+    header: props => <ColumnHeader {...props} title={'Name'} />,
+    enableColumnFilter: false,
+    sortingFn: 'alphanumericCaseSensitive',
+    enableSorting: true,
+  }),
+  columnHelper.accessor('type', {
+    cell: ctx => {
+      const entityType = convertToEntityType(ctx.row.original.type)
+      return (
+        <>
+          <EntityTypeIcon type={entityType} style={{ marginRight: '5px' }} />
+          {entityTypeToFriendlyName(entityType)}
+        </>
+      )
+    },
+    header: props => <ColumnHeader {...props} title={'Type'} />,
+    enableGlobalFilter: false,
+    sortingFn: 'alphanumeric',
+    filterFn: 'arrIncludes',
+    meta: {
+      enableMultipleSelect: true,
+      filterVariant: 'enumeration',
+      getDisplayText: value =>
+        entityTypeToFriendlyName(convertToEntityType(value)),
+    },
+    enableSorting: true,
+  }),
+]
+
+function filterEntityHeaders(searchTerm: string, array: EntityHeader[]) {
+  if (isEmpty(searchTerm)) {
+    return array
+  }
+  const searchTermLowercase = searchTerm.toLowerCase()
+  return array.filter(
+    item => item.name.toLowerCase().indexOf(searchTermLowercase) >= 0,
+  )
+}
+
 export default function FavoritesPage() {
   const { accessToken } = useSynapseContext()
-  const [sort, setSort] = useState<Sort | undefined>(undefined)
   const [searchText, setSearchText] = useState<string>('')
-  const [sortedData, setSortedData] = useState<EntityHeader[] | undefined>(
-    undefined,
-  )
   const [error, setError] = useState<Error>()
-  const { data, isFetching, isError, error: newError } = useGetFavorites()
+  const { data, isLoading, isError, error: newError } = useGetFavorites()
+
+  const favorites = useMemo(
+    () => filterEntityHeaders(searchText, data?.results ?? []),
+    [data?.results, searchText],
+  )
+
+  const favoritesTable = useReactTable<EntityHeader>({
+    data: favorites,
+    columns: columns,
+    getCoreRowModel: getCoreRowModel(),
+    getFilteredRowModel: getFilteredRowModel(),
+    getSortedRowModel: getSortedRowModel(),
+    getFacetedRowModel: getFacetedRowModel(),
+    getFacetedUniqueValues: getFacetedUniqueValues(),
```

Comment:
Added ability to filter on entity type. This client-side filter implementation is provided by TanStack Table.